docs/server/node/upgrade_guide/v5ToV6UpgradeGuide.mdx

```diff
@@ -0,0 +1,108 @@
+---
+slug: /server/nodejsServerSDK/upgrade_guide/v5ToV6
+title: Node SDK v5.x.x to v6.0.0 Upgrade Guide
+---
+### Summary 
+
+Node.js SDK V6.0.0 introduced breaking changes on 
+- Removed async version of core APIs (checkGate, getConfig..)
+- Changed DataAdapter cache keys SDK used to read and write from
+- Changed default hashing algorithm used for getClientInitializeResponse() which is being used for bootstrapping client sdk.
+- If no environment tier is being set for SDK, we will default environment to be `production` when evaluating. 
+- Changed how you should disable exposure logging
+
+#### Synchronous Core APIs
+**Changes**
+Changed all core apis to be synchronous functions 
+```javascript
+// All removed
+async getFeatureGate(): Promise<FeatureGate> // Removed
+getFeatureGate(): FeatureGate // New API
+
+async checkGate(): Promise<bool> // Removed
+checkGate(): bool // New API
+
+async getConfig(): Promise<DynamicConfig>
+getConfig(): DynamicConfig // New API
+
+async getExperiment(): Promise<DynamicConfig>
+getExperiment(): DynamicConfig // New API
+....
+
+```
+
+**Context**
+The initial design of async APIs are ensuring every evaluation can be done properly even when the SDK is not compatible by requesting evaluation from Statsig Server. 
+However, as SDK and evaluation rules become more stable, we rarely see such cases. For the performance of core apis, we make all apis synchronous functions.
+
+**Migration Guide**
+If you are using gate as following way, and you are using **javascript runtime error** will be reported, please audit and update the usage
+If you are using typescript, compile error will be reported.
+Meanwhile, we deprecated functions `*Sync()` for example `checkGateSync()`, please use checkGate() 
+```javascript
+Statsig.checkGate(user, "gate").then(() => {
+  // do somethig
+}).catch() // Compile error if you are using typescript. If you are using javascript, there will be runtime error please audit and 
+```
+If you re using as following ways, still recommend to remove await, but there won't be any error reported.
+``` javascript
+const gate = await Statsig.checkGate(user, "gate")
+```
+#### DataAdapter Cache Key Changes
+**Context**
+To better support using multiple Statsig Instances in one runtime, and also better support over different responses format, for example download_config_spec_v1 vs download_config_spec_v2, we updated the cache keys SDK read and write from to avoid cache key collision. 
+
+**Changes**
+- Change cache key for config_specs from `statsig.cache` to 
+
+ `statsig|v1/download_config_specs|{String(useGzip)}|{SHA256Hashed(secretkey)}`
```

Comment:
worth noting they should make it able to handle v2/download_config_specs as that may come soon?